tools/wifi_scanner.py

```diff
@@ -0,0 +1,177 @@
+import serial
+import time
+import json
+from typing import List, Dict
+
+class ESPWiFiScanner:
+    def __init__(self, port: str, baudrate: int = 115200):
+        self.port = port
+        self.baudrate = baudrate
+        self.serial = None
+
+    def connect(self) -> bool:
+        try:
+            self.serial = serial.Serial(
+                port=self.port,
+                baudrate=self.baudrate,
+                timeout=1
+            )
+            return True
+        except serial.SerialException as e:
+            print(f"Error connecting to ESP32: {e}")
+            return False
+
+    def scan_networks(self, timeout_seconds: int = 30) -> List[Dict]:
+        if not self.serial:
+            print("Not connected to ESP32")
+            return []
+
+        self.serial.reset_input_buffer()
+        
+        command = '{"commands":[{"command":"scan_networks","data":{}}]}\n'
```

Comment:
Small thing, `data` part should be completely optional, so we don't have to send that